src/annotator/highlighter.js

```diff
@@ -150,19 +144,48 @@ function drawHighlightsAbovePdfCanvas(highlightEl) {
  * @typedef {HTMLElement & HighlightProps} HighlightElement
  */
 
+/**
+ * Return the complete text nodes inside `range`.
+ *
+ * If a range starts or ends part-way through a text node, the nodes are split
+ * and the part inside the range is returned.
+ *
+ * @param {Range} range
+ * @return {Text[]}
+ */
+function wholeTextNodesInRange(range) {
+  const textNodes = [];
+
+  forEachNodeInRange(range, node => {
+    if (node.nodeType !== Node.TEXT_NODE) {
+      return;
+    }
+    let text = /** @type {Text} */ (node);
```

Comment:
We chatted a little in Slack: I think it would be helpful to drop a comment or two in this logic. You're using legit web APIs here, but I did have to spend a few minutes on MDN's docs about [`splitText`](https://developer.mozilla.org/en-US/docs/Web/API/Text/splitText) to understand what was going on. i.e. I know any comments here would be technically superfluous but might save someone a little time later, especially if they captured the _intent_ of the logic.

Reply:
Done. I've added comments to clarify.